### PR TITLE
Appointment feature

### PR DESCRIPTION
**Pull request by Uzair-Manzoor**
## This PR has been created against below mentioned implementations into the project:

- [x] added the form for booking an appointment
- [x] added my appointments page
- [x] fixed linters

### CODE REVIEW

File: src/redux/appointment/appointmentsSlice.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+const createAppointment = createAsyncThunk('user/createAppointment', async (data) => {
+  try {
+    const response = await fetch(url, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: `Bearer ${JSON.parse(localStorage.getItem('user'))?.token}`,
+      },
+      body: JSON.stringify(data),
+    });
```
Comment (mahdinoori2000) [inline src/redux/appointment/appointmentsSlice.js:14]: I encountered a 404 error when attempting to submit an appointment. It seems to be originating from the slice. Could you please investigate and resolve this issue?

**Review by mahdinoori2000 — CHANGES_REQUESTED** [review]
Hi @Uzair-Manzoor , 

Great job so far! 👏 You've done well, and you're almost there for the next project ⏭️, but there are a few issues 🐛 that you still need to work on.




## Highlights
✅ Great UI
✅ Clean Code



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

-[ ] Kindly ensure that add the dependencies in the PR description that you installed in the branch. It's a best practice.

File: src/components/appointmentForm/AppointmentForm.jsx [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+  const navigate = useNavigate();
+  const [selectedDoctor, setSelectedDoctor] = useState('');
+  const [doctorId, setDoctorId] = useState();
+  const [selectedCity, setSelectedCity] = useState('');
+  const [selectedDate, setSelectedDate] = useState(dayjs());
+  const [selectedTime, setSelectedTime] = useState(dayjs());
+
+  // Ensure fetchedDoctors is initialized as an array
+  const fetchedDoctors = useSelector((state) => state.doctors.doctors) || [];
+
```
Comment (mahdinoori2000) [inline src/components/appointmentForm/AppointmentForm.jsx:29]: Thank you for implementing the appointment booking feature. However, I noticed that when attempting to book an appointment, the doctor's name isn't displayed. Could you please address this issue?

now showing doctor

File: src/redux/appointment/appointmentsSlice.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+const createAppointment = createAsyncThunk('user/createAppointment', async (data) => {
+  try {
+    const response = await fetch(url, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: `Bearer ${JSON.parse(localStorage.getItem('user'))?.token}`,
+      },
+      body: JSON.stringify(data),
```
Comment (mahdinoori2000) [inline src/redux/appointment/appointmentsSlice.js:13]: I encountered a 404 error when attempting to submit an appointment. It seems to be originating from the slice. Could you please investigate and resolve this issue?

Comment (Uzair-Manzoor) [reply src/redux/appointment/appointmentsSlice.js:13]: Hi @mahdinoori2000 , thanks for your feedback. I'm working to fix the addressed issues 😭

**Pull request merged**